web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/Main.java

```diff
@@ -0,0 +1,233 @@
+                    blobInfo,
+                    15,
+                    TimeUnit.MINUTES,
+                    Storage.SignUrlOption.httpMethod(HttpMethod.PUT),
+                    Storage.SignUrlOption.withExtHeaders(extensionHeaders),
+                    Storage.SignUrlOption.withV4Signature());
+
+        setJobCountryCode(jobId, countryCode);
+        return "{\"jobId\": \"" + jobId + "\", \"url\": \"" + url.toString() + "\"}";
+      }
+    } catch (Exception exc) {
+      System.out.println("Error:" + exc.toString());
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Error", exc);
+    }
+  }
+
+  @PostMapping("/run-validator")
+  public ResponseEntity runValidator(@RequestBody Body body) {
+    try {
+      var message = body.getMessage();
+      if (message == null) {
+        var msg = "Bad Request: invalid Pub/Sub message format";
+        System.out.println(msg);
+        return new ResponseEntity(msg, HttpStatus.BAD_REQUEST);
+      }
+
+      var data = new String(Base64.getDecoder().decode(message.getData()));
+
+      var map = new ObjectMapper();
+      var node = map.readTree(data);
+
+      var inputFilename = node.get("name").textValue();
+      var jobId = inputFilename.split("/")[0];
+      var tempDir = Files.createTempDirectory(tempFolderName).toFile();
+
+      var inputResource =
+          applicationContext.getResource("gs://" + userUploadBucketName + "/" + inputFilename);
+      var inputFile = inputResource.getInputStream();
+
+      var tempFile = File.createTempFile(jobId, ".zip", tempDir);
+      var output = new FileOutputStream(tempFile);
+      inputFile.transferTo(output);
+
+      var countryCode = getJobCountryCode(jobId);
+
+      var runner = new ValidationRunner(new VersionResolver());
+      var outputPath = new File(tempDir.toPath().toString() + jobId);
+      var config =
+          ValidationRunnerConfig.builder()
+              .setGtfsSource(tempFile.toURI())
+              .setOutputDirectory(outputPath.toPath())
+              .setCountryCode(CountryCode.forStringOrUnknown(countryCode))
+              .build();
+      runner.run(config);
+
+      var directoryListing = outputPath.listFiles();
+      if (directoryListing != null) {
+        for (var reportFile : directoryListing) {
+          System.out.println(
+              ("Writing to : "
+                  + "gs://"
+                  + resultsBucketName
+                  + "/"
+                  + jobId
+                  + "/"
+                  + reportFile.getName()));
+
+          var blobId = BlobId.of(resultsBucketName, jobId + "/" + reportFile.getName());
+          var mimeType = Files.probeContentType(reportFile.toPath());
+          var blobInfo = BlobInfo.newBuilder(blobId).setContentType(mimeType).build();
+          var fileBytes = Files.readAllBytes(Paths.get(reportFile.getPath()));
+          storage.create(blobInfo, fileBytes);
+        }
+      }
+
+      tempDir.delete();
```

Comment:
Does there need to be some sort of refactoring to make sure this gets called even if intermediate code throws an exception?